src/components/GraphiQL.js

```diff
@@ -475,7 +475,13 @@ export class GraphiQL extends React.Component {
 
   _fetchQuery(query, variables, operationName, cb) {
     const fetcher = this.props.fetcher;
-    const fetch = fetcher({ query, variables, operationName });
+    const parsedVariables =
+      variables && variables.trim() !== '' ? JSON.parse(variables) : null;
```

Comment:
I definitely share the concern about backward compatibility. One possibility would be to add an extra step in the migration path and initially support a prop that configures variables format (string/json). This will make the migration a little bit more graceful, but still it would be a breaking change. Though I wonder how many servers do not support JSON object variables and support only string.